arches/app/views/workflow_history.py

```diff
@@ -60,8 +58,8 @@ def post(self, request, workflowid):
                         workflowid = workflowid,
                     )
                 else:
-                    return JSONErrorResponse(_("Request Failed"), _("Permission Denied"), status=403)
-            
+                    raise PermissionDenied
```

Comment:
Will the UI be able to handle the fact that we used to return a JSON object and are now returning a python exception?

Reply:
Good question. I just retested. No change on the frontend because the frontend doesn't even check the response for success:

Notice no `try/catch` here:

https://github.com/archesproject/arches/blob/a1c2b429f5aaa54af95a4111862eb87d56709107/arches/app/media/js/views/components/workflows/workflow-component-abstract.js#L776

This is probably because you shouldn't be able to end up in this scenario (editing someone else's workflow) through the UI. You'd need to be trying to hack the endpoint with your own POST requests to reach this. I tested by leaving the UI open and then changing the owner of the workflow in the database.

In terms of the motivation for this change, I figured it was better to have all the permission failure cases return the same shape of a response so that we don't expose any information about what workflows exist.

Reply:
what about here?
https://github.com/archesproject/arches/blob/dev/7.6.x/arches/app/media/js/viewmodels/workflow.js#L371-L394

Reply:
Retested this one also. Same. Both 7.6 and this PR show the same in the UI:

![Screenshot 2024-03-14 at 8 26 02 PM](https://github.com/archesproject/arches/assets/38668450/8fc51d1b-3fbc-4fee-9395-2f5ea1caae27)

The fact that the response changes from:

dev/7.6.x
```
{
    "message": "Permission Denied",
    "status": "false",
    "success": false,
    "title": "Request Failed"
}
```
to
```
<!doctype html>
<html lang="en">
<head>
  <title>403 Forbidden</title>
</head>
<body>
  <h1>403 Forbidden</h1><p></p>
</body>
</html>
```

doesn't seem to matter.

Reply:
All we're using is the statusText which remains the same.